exchanges/kucoin/kucoin_websocket.go

```diff
@@ -0,0 +1,1882 @@
+		Price:           resp.OrderPrice,
+		Exchange:        ku.Name,
+		ExecutedAmount:  resp.FilledSize,
+		RemainingAmount: resp.RemainSize,
+		ClientOrderID:   resp.ClientOid,
+		OrderID:         resp.TradeID,
+		AssetType:       asset.Futures,
+		LastUpdated:     resp.OrderTime.Time(),
+	}
+	return nil
+}
+
+func (ku *Kucoin) processFuturesTransactionStatistics(respData []byte, instrument string) error {
+	resp := WsFuturesTransactionStatisticsTimeEvent{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	resp.Symbol = instrument
+	return nil
+}
+
+func (ku *Kucoin) processFuturesSystemAnnouncement(respData []byte, subject string) error {
+	resp := WsFuturesFundingBegin{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	resp.Subject = subject
+	ku.Websocket.DataHandler <- &resp
+	return nil
+}
+
+func (ku *Kucoin) processFuturesFundingData(respData []byte, instrument string) error {
+	resp := WsFundingRate{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	resp.Symbol = instrument
+	ku.Websocket.DataHandler <- &resp
+	return nil
+}
+
+func (ku *Kucoin) processFuturesMarkPriceAndIndexPrice(respData []byte, instrument string) error {
+	resp := WsFuturesMarkPriceAndIndexPrice{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	resp.Symbol = instrument
+	ku.Websocket.DataHandler <- &resp
+	return nil
+}
+
+func (ku *Kucoin) processFuturesOrderbookLevel5(respData []byte, instrument string) error {
+	response := WsOrderbookLevel5Response{}
+	if err := json.Unmarshal(respData, &response); err != nil {
+		return err
+	}
+	resp := response.ExtractOrderbookItems()
+	cp, err := currency.NewPairFromString(instrument)
+	if err != nil {
+		return err
+	}
+	return ku.Websocket.Orderbook.Update(&orderbook.Update{
+		UpdateID:   resp.Sequence,
+		UpdateTime: resp.Timestamp.Time(),
+		Asset:      asset.Futures,
+		Bids:       resp.Bids,
+		Asks:       resp.Asks,
+		Pair:       cp,
+	})
+}
+
+func (ku *Kucoin) processFuturesOrderbookLevel2(respData []byte, instrument string) error {
+	resp := WsFuturesOrderbokInfo{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	detail, err := ku.GetFuturesPartOrderbook100(context.Background(), instrument)
+	if err != nil {
+		return err
+	}
+	pair, err := currency.NewPairFromString(instrument)
+	if err != nil {
+		return err
+	}
+	base := orderbook.Update{
+		UpdateTime: detail.Time,
+		Pair:       pair,
+		Asset:      asset.Futures,
+		Asks:       detail.Asks,
+		Bids:       detail.Bids,
+	}
+	return ku.Websocket.Orderbook.Update(&base)
+}
+
+func (ku *Kucoin) processFuturesTickerV2(respData []byte) error {
+	resp := WsFuturesTicker{}
+	if err := json.Unmarshal(respData, &resp); err != nil {
+		return err
+	}
+	enabledPairs, err := ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		return err
+	}
+	pair, err := currency.NewPairFromString(resp.Symbol)
```

Comment:
For futures based websocket endpoints, please change this to:
```go
pair, err = enabledPairs.DeriveFrom(resp.Symbol)
```
Because right now this isn't working properly:
<img width="1133" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9261323/3dc31e0d-db88-45a2-a543-7be2da0fba24">